cpplint.py

```diff
@@ -5484,6 +5485,19 @@ def CheckLanguage(filename, clean_lines, linenum, file_extension,
           'https://google-styleguide.googlecode.com/svn/trunk/cppguide.xml#Namespaces'
           ' for more information.')
 
+  # Check namespace names for correct naming, according to
+  # https://google.github.io/styleguide/cppguide.html#Namespace_Names
+  # namespace names are "all lower-case, with words separated by underscores"
+  match = re.match(r'.*namespace\s+([^{=;]+)\s+[{;=]', line)
+  if match:
+    name = match.group(1)
+    if name != "" and not re.match(r'^((inline )?[a-z_]+(::)?)+$', name):
```

Comment:
It would be much better if the detection was based on nesting_state. The could changed by #235 should provide an example.